### ISSUE TITLE
Screen Reader Headings Improvements Test Plan 4.0

### ISSUE THREAD

**Issue by saad-mo** [SAP/spartacus#12888]
## Headings Changes [Master Ticket: #11192]

- Full details of the changes can be found on [this page](https://cxwiki.sap.com/pages/viewpage.action?spaceKey=spar&title=Headings+and+Labels). 
- These changes were made in order to group headings elements correctly and with the appropriate contextual hierarchy.
- Changes mostly target B2C storefront.
- Part 2 of the master ticket `Lables` is still in progress 
- Please verify that headings match with the changes done for each page from the list below.

### Testing with Screen Reader on Mac using VoiceOver
- VoiceOver is a native Screen Reader installed by default on Mac
- To bring up VoiceOver screen reader, press `cmd` + `f5`
- Make sure the Spartacus page that you are testing is in focus by clicking, tabbing to the page elements 
- Bring up `Rotor` feature of VoiceOver by pressing `ctrl`+`optn`+`u`
- Use `left` or `right` arrow keys to navigate to `Headings` list if it’s not the default lost (Refer to example screen shot below)
<img width="1157" alt="2021-06-22_14-55-51" src="https://user-images.githubusercontent.com/15215400/122991189-5de1b800-d36a-11eb-8158-cb32aef3fb6e.png">

### Testing with Screen Reader on Windows using JAWS
- To bring up the screen reader, click on its Windows icon and to close click on its window closing button.
- Make sure the Spartacus page that you are testing is in focus by clicking, tabbing to the page elements 
- Bring up the `Headings List` dialog by pressing `insert`+`f6`
- Use `up` or `down` arrow keys to navigate through Headings.
<img width="960" alt="114077383-bd551f00-9875-11eb-849b-0aec45692421" src="https://user-images.githubusercontent.com/15113219/123172977-e97f4580-d44b-11eb-8d79-eb69ae666874.png">

 
### Testing with Screen Reader on Windows using NVDA
- To bring up the screen reader, click on its Windows icon and close press `insert`+`q`.
- Make sure the Spartacus page that you are testing is in focus by clicking, tabbing to the page elements 
- Bring up the `Elements List` dialog by pressing `insert`+`f7`
- Use the `tab` to reach the first option (radio button) and `right` arrow key to navigate to the Headings option, `tab` to go into it, then `up` or `down` arrow keys to navigate through Headings.
<img width="960" alt="114077078-694a3a80-9875-11eb-9249-2389fffc7ec0" src="https://user-images.githubusercontent.com/15113219/123173017-f6039e00-d44b-11eb-8951-322260db42cf.png">

#### Global Header 
- [x] Removed `h5` tags from top nav link 

### Global Footer 
- [x] Removed `h5` headings from footer links headings

### Homepage 
 - [x] `h1` Homepage tag added, this tag should be visually hidden 
 - [x] `h2` Best Selling Products, What's New (previously `h3`)
 
### Product Details Page 
- [x] `h1` Product Title
- [ ] `h2` Tab text **(Dropping this one as it should be vocalized as a tab button)**
- [x] `h3` All headings under tab content 

### Search Page 
- [x] `h1` Search Results 
- [x] `h2` Product Title 

### Cart Page 
- [x] `h1` Your Shopping Cart 
- [x] `h2` Product Titles 
- [x] `h3` ORDER SUMMARY (previously `h4`)
- [x] `h4` Cart #ID

### Checkout Pages 
- [x] `h1` Checkout # Items 
- [x] `h2` Each Step Heading: Shipping Address, Shipping Method, Payment, Review (previously `h3`) 
- [x] `h3` On Review Step: Payment, Ship to, Shipping Method
- [x] `h4` Cart Total (X items): $xxx.xx

### Store Finder Page
- [x] `h1` Store Finder 
- [x] `h2` All counties with a number, All location titles once you click on the country detail 

### Whishlist Page 
- [x] `h1` Wish List 
- [x] `h2` No product on your wish list yet (when empty), Product titles (non-empty page) 

---

#### Global Header 
- [x] Removed `h5` tags from top nav link 

### Global Footer 
- [x] Removed `h5` headings from footer links headings

### Homepage 
 - [ ] `h1` Homepage tag added, this tag should be visually hidden 
 - [x] `h2` Best Selling Products, What's New (previously `h3`)
 
### Product Details Page 
- [x] `h1` Product Title
- [ ] `h2` Tab text **(Dropping this one as it should be vocalized as a tab button)**
- [x] `h3` All headings under tab content 

### Search Page 
- [x] `h1` Search Results 
- [x] `h2` Product Title 

### Cart Page 
- [x] `h1` Your Shopping Cart 
- [x] `h2` Product Titles 
- [x] `h3` ORDER SUMMARY (previously `h4`)
- [x] `h4` Cart #ID

### Checkout Pages 
- [x] `h1` Checkout # Items 
- [x] `h2` Each Step Heading: Shipping Address, Shipping Method, Payment, Review (previously `h3`) 
- [x] `h3` On Review Step: Payment, Ship to, Shipping Method
- [x] `h4` Cart Total (X items): $xxx.xx

### Store Finder Page
- [x] `h1` Store Finder 
- [x] `h2` All counties with a number, All location titles once you click on the country detail 

### Whishlist Page 
- [x] `h1` Wish List 
- [x] `h2` No product on your wish list yet (when empty), Product titles (non-empty page)

**Comment by giancorderoortiz** [comment]
Is mobile testing included here? Is it a business case?

**Comment by developpeurweb** [comment]
> Is mobile testing included here? Is it a business case?

I'll include mobile testing for Android and iPhone, although these won't be blockers for now. 
No business case, it should be considered for the future, but not at this point since these updates are still very minimal under the full extent of the Screen Reader milestone.

**Comment by developpeurweb** [comment]
Test passed successfully. Small issues found: specific tickets for each will be created.
